source/OrderedTypes/SupLattice.lagda

```diff
@@ -0,0 +1,286 @@
+Ian Ray, 05/02/2024.
+
+A Sup Lattice L is a set with a partial order ≤ that has suprema of 'small'
+types. We will use three universe parameters: 𝓤 for the carrier, 𝓦 for the
+order values and 𝓥 for the families which have suprema. 
+
+\begin{code}
+
+{-# OPTIONS --safe --without-K --exact-split #-}
+
+open import MLTT.Spartan
+open import UF.Equiv
+open import UF.Equiv-FunExt
+open import UF.EquivalenceExamples
+open import UF.FunExt
+open import UF.Hedberg
+open import UF.Logic
+open import UF.Powerset-MultiUniverse
+open import UF.PropTrunc
+open import UF.Retracts
+open import UF.Sets
+open import UF.Subsingletons
+open import UF.Subsingletons-FunExt
+open import UF.SubtypeClassifier
+open import UF.Size
+open import UF.SmallnessProperties
+open import UF.UniverseEmbedding
+
+module OrderedTypes.SupLattice
+       (pt : propositional-truncations-exist)
+       (fe : Fun-Ext)
+        where
+
+open import Locales.Frame pt fe hiding (⟨_⟩ ; join-of)
+open import Slice.Family
+open import UF.ImageAndSurjection pt
+
+open AllCombinators pt fe
+open PropositionalTruncation pt
+
+\end{code}
+
+We commence by defining sup lattices and some boiler plate. 
+
+\begin{code}
+
+module sup-lattice-def (𝓤 𝓦 𝓥 : Universe) where
+
+ sup-lattice-data : 𝓤  ̇ → 𝓤 ⊔ 𝓦 ⁺ ⊔ 𝓥 ⁺  ̇
+ sup-lattice-data A = (A → A → Ω 𝓦) × (Fam 𝓥 A → A)
+ 
+ is-sup-lattice : {A : 𝓤  ̇} → sup-lattice-data A → 𝓤 ⊔ 𝓦 ⊔ 𝓥 ⁺  ̇  
+ is-sup-lattice {A} (_≤_ , ⋁_) = is-partial-order A _≤_ × suprema
+  where
+   open Joins _≤_
+   suprema : 𝓤 ⊔ 𝓦 ⊔ 𝓥 ⁺  ̇
+   suprema = (U : Fam 𝓥 A) → ((⋁ U) is-lub-of U) holds
+
+ sup-lattice-structure : 𝓤 ̇ → 𝓤 ⊔ 𝓥 ⁺ ⊔ 𝓦 ⁺ ̇
+ sup-lattice-structure A = Σ d ꞉ (sup-lattice-data A) , is-sup-lattice d
+
+Sup-Lattice : (𝓤 𝓦 𝓥 : Universe) → (𝓤 ⊔ 𝓦 ⊔ 𝓥) ⁺  ̇
```

Comment:
You can simply make `sup-lattice-def` anonymous and move `Sup-Lattice` into the module.

Reply:
Do you just want me to move Sup-Lattice in or all the boiler plate as well?

Reply:
I will leave boiler plate out since you didn't mention it and I don't want to give universe params. for all of the boiler plate.